### PR TITLE
Expose collector ports in docker images

### LINKED ISSUE #4808: [Feature]: Expose OTLP Ports in all-in-one docker image
### Requirement

In the all-in-one docker image Open Telemetry protocol ports (`4317`, `4318`) need to be exposed. https://github.com/jaegertracing/jaeger/blob/main/cmd/all-in-one/Dockerfile

### Problem

We have a testcontainer setup where we rely on the ports marked as exposed by the image. Eg. when running the image with "publish all" flag ie. `docker run -P` the OTLP ports are not published 

### Proposal

Let me know if this is okay to expose. Will raise PR post discussion.

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by arunvelsriram**
## Which problem is this PR solving?
Resolved #4808 

## Description of the changes
- Expose OTLP gRPC and HTTP Collector ports in all-in-one Docker image.
- Expose all collector ports in collector Docker image

## How was this change tested?
- 

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by yurishkuro** [comment]
do you want to fix cmd/collector/Dockerfile too? It's missing most ports

**Comment by arunvelsriram** [comment]
Shall I expose all [collector ports](https://www.jaegertracing.io/docs/1.6/deployment/#collectors) in the collector docker file as part of this PR?

**Comment by arunvelsriram** [comment]
> do you want to fix cmd/collector/Dockerfile too? It's missing most ports

done

File: cmd/collector/Dockerfile [hunk 1]
```diff
@@ -5,14 +5,14 @@ FROM $base_image AS release
...
+EXPOSE 14250/tcp 14268/tcp 9411/tcp 4317/tcp 4318/tcp
 ENTRYPOINT ["/go/bin/collector-linux"]
 USER ${USER_UID}
 
 FROM $debug_image AS debug
 ARG TARGETARCH=amd64
 ARG USER_UID=10001
 COPY collector-debug-linux-$TARGETARCH /go/bin/collector-linux
-EXPOSE 12345/tcp 14250/tcp
+EXPOSE 12345/tcp 14250/tcp 14268/tcp 9411/tcp 4317/tcp 4318/tcp
```
Comment (arunvelsriram) [inline cmd/collector/Dockerfile:16]: Is `12345/tcp` required?

File: cmd/collector/Dockerfile [hunk 2]
```diff
@@ -5,14 +5,14 @@ FROM $base_image AS release
 ARG TARGETARCH
 ARG USER_UID=10001
 COPY collector-linux-$TARGETARCH /go/bin/collector-linux
-EXPOSE 14250/tcp
+EXPOSE 14250/tcp 14268/tcp 9411/tcp 4317/tcp 4318/tcp
```
Comment (yurishkuro) [inline cmd/collector/Dockerfile:8]: it would be nice to do that in the same style as all-in-one, with comments explaining each port

Comment (arunvelsriram) [reply cmd/collector/Dockerfile]: Makes sense. I have pushed the changes

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
thanks 🎉

**Pull request merged**